code/script.js

```diff
@@ -222,104 +226,127 @@ const generateBoard = () => {
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  } else {
+    if (keep) { //hair and eyes
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+      alert(`Yes! The person has ${value} ${category}! Let's keep all people with ${value} ${category}.`)
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+    } else {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+      alert(`No, the person doesn't have ${value} ${category}! Let's remove all people with ${value} ${category}.`)
+
+    }
+  }
+  generateBoard(); //generates the board after filtering.
 
-  // Invoke a function to redraw the board with the remaining people.
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const isConfirmed = confirm(`Are you sure you want to guess that ${personToConfirm} is the secret person?`);
+
+  if (isConfirmed) {
+    checkMyGuess(personToConfirm)
+  }
 }
 
-// If you confirm, this function is invoked
+// If player confirms, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {
+    winOrLose.style.display = "flex";
+    board.style.display = 'none';
+    winOrLoseText.innerHTML += `<h1>Congratulations! Your guess is right and you WON! 🏆</h1>`
+  } else {
+    winOrLose.style.display = "flex";
+    board.style.display = 'none';
+    winOrLoseText.innerHTML += `<h1>Your guess is unfortunately wrong 😞 Do you wanna try again?</h1>`
+  }
+
 }
 
 // Invokes the start function when website is loaded
-start()
+start();
+
+const replay = () => {
+  location.reload(); //The location.reload refreshes the page, which was a easy solution to make the game start again.
+}
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+restartButton.addEventListener('click', start); //resets the board when the restart button is clicked.
+
+questions.addEventListener('click', selectQuestion); //Does this work? Error in console on website.
```

Comment:
im also wondering about this, doesn't select question get called when you click the findout button?